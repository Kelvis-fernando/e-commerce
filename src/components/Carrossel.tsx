import { CaretLeft, CaretRight } from "phosphor-react";
import React from "react";
import { useCarrossel } from "../hooks/useCarrossel";
import {
  CarouselWrapper,
  LeftArrow,
  RightArrow,
  CarouselImage,
} from "../styles/carrossel";

interface CarrosselProps {
  images: string[];
}

const Carrossel: React.FC<CarrosselProps> = ({ images }) => {
  const { handlePrevClick, handleNextClick, currentImageIndex } =
    useCarrossel(images);

  return (
    <CarouselWrapper>
      <LeftArrow onClick={handlePrevClick}>
        <CaretLeft size={32} />
      </LeftArrow>
      <RightArrow onClick={handleNextClick}>
        <CaretRight size={32} />
      </RightArrow>
      <CarouselImage    
        src={images[currentImageIndex]}
        alt={`Image ${currentImageIndex + 1}`}
      />
    </CarouselWrapper>
  );
};

export default Carrossel;
