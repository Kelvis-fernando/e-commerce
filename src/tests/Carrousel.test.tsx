import React from "react";
import { render } from "@testing-library/react";
import Carrossel from "../components/Carrossel";

describe("Carrossel", () => {
  test("should render and match snapshot", () => {
    const { container } = render(<Carrossel images={[]} />);

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
