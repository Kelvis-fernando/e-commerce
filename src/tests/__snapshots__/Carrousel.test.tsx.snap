// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carrossel should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe iYlegs"
  >
    <div
      class="sc-dkrFOg WKixe"
    >
      <svg
        fill="currentColor"
        height="32"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <polyline
          fill="none"
          points="160 208 80 128 160 48"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </div>
    <div
      class="sc-hLBbgP jHwDjN"
    >
      <svg
        fill="currentColor"
        height="32"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="256"
          width="256"
        />
        <polyline
          fill="none"
          points="96 48 176 128 96 208"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="16"
        />
      </svg>
    </div>
    <img
      alt="Image 1"
      class="sc-gswNZR cfXGIm"
    />
  </div>
</div>
`;
